### PR TITLE
[SYCL-MLIR] Remove typed pointer support

### PR DESCRIPTION
**Pull request by sommerlukas**
With the pulldown in #11160, SYCL-MLIR received the commits completing opaque pointer support SYCL, making typed pointers obsolete. 

This removes the typed pointer support from the SYCL-MLIR compiler, to reduce the maintenance burden, as it is now obsolete.

### CODE REVIEW

**Review by whitneywhtsang — APPROVED** [review]

File: polygeist/include/mlir/Dialect/Polygeist/Transforms/Passes.h [hunk 1]
```diff
@@ -54,8 +53,6 @@ std::unique_ptr<Pass> createLoopRestructurePass();
 std::unique_ptr<Pass> createMem2RegPass();
 std::unique_ptr<Pass> createOpenMPOptPass();
 std::unique_ptr<Pass> createParallelLowerPass();
-std::unique_ptr<Pass>
-createParallelLowerPass(const ParallelLowerOptions &options);
```
Comment (victor-eds) [inline polygeist/include/mlir/Dialect/Polygeist/Transforms/Passes.h:58]: Why are we removing this?

File: mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp [hunk 2]
```diff
@@ -97,7 +97,7 @@ void ConvertSYCLToLLVMPass::runOnOperation() {
 
   LowerToLLVMOptions options(&context);
   options.useBarePtrCallConv = true;
-  options.useOpaquePointers = useOpaquePointers;
+  options.useOpaquePointers = true;
```
[the remark this answers is not in the document]
Comment (sommerlukas) [reply mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp:100]: I think the upstream LLVM dialect has not yet entirely removed the typed pointer support, so setting this here to make sure we get the behavior.

[the remark this answers is not in the document]
Comment (victor-eds) [reply mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp:100]: Yes, I checked they still have some typed stuf. Thanks!

File: polygeist/include/mlir/Dialect/Polygeist/Transforms/Passes.h (hunk 1 above)
Comment (victor-eds) [reply polygeist/include/mlir/Dialect/Polygeist/Transforms/Passes.h:58]: Oh, missed that. Thanks!

**Review by victor-eds — APPROVED** [review]

**Review by etiotto — COMMENTED** [review]
LGTM

**Pull request merged**